lib/spack/spack/installer.py

```diff
@@ -2224,11 +2219,6 @@ def install(self) -> None:
                 if not keep_prefix and not action == InstallAction.OVERWRITE:
                     pkg.remove_prefix()
 
-                # The subprocess *may* have removed the build stage. Mark it
-                # not created so that the next time pkg.stage is invoked, we
-                # check the filesystem for it.
-                pkg.stage.created = False
```

Comment:
This is redundant because `restage` is true by default, so we hit `stage.destroy()` in `BuildProcessInstaller.run()`